Sources/SpeziAccount/AccountValue/Configuration/AccountKeyRequirement.swift

```diff
@@ -0,0 +1,26 @@
+//
+// This source file is part of the Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// Describes the user-configured requirement level for an ``AccountKey``.
+public enum AccountKeyRequirement: Sendable {
+    /// The associated account value **must** be provided by the user at signup.
+    ///
+    /// It is mandatory to use the ``RequiredAccountKey`` protocol.
+    case required
```

Comment:
One element that I am debating here is the following use case:
We have one application that uses an invitation-code bases system for signing up. The users have to enter their DOB and Invitation code that is sent to them via email that is already configured. How would you classify this. Probably required, but not something that should be editable in the overview as it is no longer relevant once it it collected.

Reply:
Using the current model you would probably model it similarly to the `PasswordKey`. You would configure it `required` on signup but you would never store it in the AccountDetails but only ever in the `SignupDetails`.
So the flow looks like that: the `SignupForm` collects the invitation code in the `SignupDetails` sent to the account service as it is marked as required. The AccountService will never set this value in the `AccountDetails` thought.
However, this only when using the `credentials` category and relying on the how `SecurityOverview` is currently implemented: https://github.com/StanfordSpezi/SpeziAccount/blob/c3e4300a1486cae0eda552569babc5ddcf3681e6/Sources/SpeziAccount/Views/AccountOverview/SecurityOverview.swift#L47-L50.

So, you are right. That currently doesn't fit very well in the model space. And the behavior isn't really well documented. Maybe we should consider a (better) way of controlling what is displayed in the `Overview`. There might be a general use case for read-only account values?

Would it make sense to track this as a separate issue or still tackle this in the PR?

Reply:
Yes, let's document and tackle this in a separate PR. The way of using a `.required` key and then not storing it in the `AccountDetails` should work for now and I like the idea of further investigating this issue when focusing on how this type of data is presented or not presented in the `Overview` 👍

Reply:
I think it would make sense to leave the user configuration model as is (the user specifying the requiredness level of an account value) and finding a way for a `AccountKey` implementation to specify if it is `readOnly`, `writeOnly` or `readWrite`. Maybe we can solve this by allowing to set something like the `EmptyView` (we probably can't use Never due to the initializer constraint) to be set for either the `DataDisplay` or `DataEntry` associated types.

Maybe I can still tackle this in this PR this way.